ChurchCRM/CRM
# Api: Person/Persons Code Cleanup - Part 1

#### What's this PR do?
- update roles to check permission
- created a new personId -> person middleware
- created a new route for person APIs  
- moved emails and roles into a persons routes
- update path to the API and updated JS files to match new API Path 

#### What are the relevant tickets?

part of #4013 #4014  


#### How should this be manually tested?
person page, click edit role, ensure the correct roles are listed and you can save the role to the person.
